sources/packages/backend/libs/integrations/src/services/disbursement-schedule/e-cert-calculation/e-cert-calculation-process.ts

```diff
@@ -128,12 +139,19 @@ export abstract class ECertCalculationProcess {
               disbursementLog,
             );
             if (!shouldProceed) {
-              // Create notifications to the student and ministry when the disbursement is blocked.
-              await this.createDisbursementBlockedNotifications(
-                eCertDisbursement.studentId,
-                eCertDisbursement.applicationNumber,
-                entityManager,
-              );
+              const shouldCreateNotification =
+                await this.shouldCreateDisbursementBlockedNotification(
+                  eCertDisbursement.disbursement.id,
+                  entityManager,
+                );
+              if (shouldCreateNotification) {
+                await this.createDisbursementBlockedNotifications(
```

Comment:
I think we should probably add back the comment "// Create notifications to the student and ministry when the disbursement is blocked." for `await this.createDisbursementBlockedNotifications`.